consensus/state_processing/src/common/initiate_validator_exit.rs

```diff
@@ -8,10 +8,10 @@ pub fn initiate_validator_exit<T: EthSpec>(
     index: usize,
     spec: &ChainSpec,
 ) -> Result<(), Error> {
-    // Return if the validator already initiated exit
-    if state.get_validator(index)?.exit_epoch != spec.far_future_epoch {
-        return Ok(());
-    }
+    // We do things in a slightly different order to the spec here. Instead of immediately checking
+    // whether the validator has already exited, we instead prepare the exit cache and compute the
+    // cheap-to-calculate values from that. *Then* we look up the validator a single time in the
+    // validator tree (expensive), make the check and mutate as appropriate.
 
```

Comment:
I don't understand how this reordering (building the exit cache first and THEN checking to see if the validator has exited) is an improvement here unless you're just trying to ensure the exit cache gets built for some later use by the calling function?

Reply:
It saves a couple of extra lookups of `state.get_validator(index)`, which are becoming more expensive with `tree-states`. I extended the comment in https://github.com/sigp/lighthouse/pull/5386